traits/tests/test_has_required_traits.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+from traits.api import Int, Float, String, HasRequiredTraits, TraitError
+
+class TestHasRequiredTraits(unittest.TestCase):
+
+    def test_trait_value_assignment(self):
+        test_instance = RequiredTest(i_trait=4, f_trait=2.2, s_trait="test")
+        self.assertEqual(test_instance.i_trait, 4)
+        self.assertEqual(test_instance.f_trait, 2.2)
+        self.assertEqual(test_instance.s_trait, "test")
+        self.assertEqual(test_instance.non_req_trait, 4.4)
+
+
+    def test_missing_required_trait(self):
+        with self.assertRaises(TraitError) as exc:
+            test_instance = RequiredTest(i_trait=3)
+        self.assertEqual(
+            exc.exception.args[0], "The following required traits were not "
+            "passed as a keyword argument: f_trait, s_trait."
```

Comment:
As it stands, the order is undefined here, so the exception message might contain s_trait, f_trait instead of f_trait, s_trait. It may actually make sense to use a sorted call when constructing the exception message, just so that it's consistent.

Reply:
That's exactly what happened on one of the builds just now!